packages/cli-kit/src/private/node/ui/components/List.tsx

```diff
@@ -25,23 +34,57 @@ const List: FunctionComponent<ListProps> = ({
   color,
   bullet = DOT,
 }): JSX.Element => {
+  function isCustomListItem(item: TokenItem<InlineToken> | CustomListItem): item is CustomListItem {
+    return (item as CustomListItem).item !== undefined
+  }
+
+  function resolveListItem(item: ListItem, index: number) {
+    const resolvedItem = {
+      index,
+      color,
+      bullet,
+      ordered,
+      item: item as TokenItem<InlineToken>,
+    }
+    return isCustomListItem(item)
+      ? {
+          ...resolvedItem,
+          ...item,
+        }
```

Comment:
`resolvedItem` already includes `item` no? why do you need to merge them here? 🤔